### PR TITLE
[shopsys] Functional documentation cart.md

### PR DESCRIPTION
**Pull request by PetraPiprekova**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Description of cart behavior.
|New feature| No 
|BC breaks| No 
|Fixes issues| ... 
|Standards and tests pass| Yes/No
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: docs/functional/wip-cart.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+Product information displayed in the cart:
+- Thumbnail
+- Name with link to product detail
+- Unit price including VAT
+- VAT rate
+- Amount
+- Price including VAT - summed price for amount of added product
+
+**Price**
```
Comment (MattCzerner) [inline docs/functional/wip-cart.md]: In other functional documentation, this is usually heading. We should make it consistent :)

File: docs/functional/wip-cart.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+- total price including VAT is automatically recalculated if the product is removed or the amount is changed
+
+**Discount code**
```
Comment (MattCzerner) [inline docs/functional/wip-cart.md]: In other functional documentation, this is usually heading. We should make it consistent :)

File: docs/functional/wip-cart.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+- discount code can be created	 in the section pricing -> promo codes in the administration
+- the input for entering the discount code is in the cart on the Front-end
+- after entering the discount code in the cart, total price with VAT is recalculated using the button "Apply"
+- discount code can be removed from the cart and total price with VAT will be recalculated automatically
+
+**Cart behavior when logging in, logging out and register the user:**
```
Comment (MattCzerner) [inline docs/functional/wip-cart.md]: In other functional documentation, this is usually heading. We should make it consistent :)

File: docs/functional/wip-cart.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+- Unit price including VAT
+- VAT rate
+- Amount
+- Price including VAT - summed price for amount of added product
+
+**Price**
+- total price including VAT is automatically recalculated if the product is removed or the amount is changed
+
+**Discount code**
+- discount code can be created	 in the section pricing -> promo codes in the administration
```
Comment (MattCzerner) [inline docs/functional/wip-cart.md]:
```suggestion
- discount code can be created in the section pricing -> promo codes in the administration
```

File: docs/functional/wip-cart.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# Cart
+This article describes functional behavior of a Cart. Article shortly summarize its Discount Codes, behavior of Cart content when user logs in and others.
```
Comment (MattCzerner) [inline docs/functional/wip-cart.md]:
```suggestion
This article describes functional behavior of a Cart. Article shortly summarize its discount codes, behavior of cart content when user logs in and others.
```

Comment (PetraPiprekova) [reply docs/functional/wip-cart.md]: fixed

File: docs/functional/wip-cart.md (hunk 1 above)
Comment (PetraPiprekova) [reply docs/functional/wip-cart.md]: ok, fixed

File: docs/functional/wip-cart.md (hunk 2 above)
Comment (PetraPiprekova) [reply docs/functional/wip-cart.md]: ok, fixed

File: docs/functional/wip-cart.md (hunk 4 above)
Comment (PetraPiprekova) [reply docs/functional/wip-cart.md]: fixed

File: docs/functional/wip-cart.md (hunk 3 above)
Comment (PetraPiprekova) [reply docs/functional/wip-cart.md]: ok, fixed

File: docs/functional/cart.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# Cart
```
Comment (LukasHeinz) [inline docs/functional/cart.md]: I miss in the article the information about the notification when product price is changed or is not available any more.

File: docs/functional/cart.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+This article describes functional behavior of a cart. Article shortly summarize its discount codes, behavior of cart content when user logs in and others.
+
+The cart is the first step of order and can be in these states:
+- **empty** - nothing was added to the cart yet, the customer is informed that the cart is empty
+- **contains goods** - the customer can see a list of products which have been added to the cart, customer can change quantity of products.
```
[the remark this answers is not in the document]
Comment (PetraPiprekova) [reply docs/functional/cart.md]: Changed

File: docs/functional/cart.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+Product information displayed in the cart:
```
[the remark this answers is not in the document]
Comment (PetraPiprekova) [reply docs/functional/cart.md]: Ok, I removed this list.

File: docs/functional/cart.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+- Thumbnail
+- Name with link to product detail
+- Unit price including VAT
+- VAT rate
+- Amount
+- Price including VAT - summed price for amount of added product
+
+## Price
+- total price including VAT is automatically recalculated if the product is removed or the amount is changed
```
[the remark this answers is not in the document]
Comment (PetraPiprekova) [reply docs/functional/cart.md]: The information is added.

File: docs/functional/cart.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+## Discount code
```
[the remark this answers is not in the document]
Comment (PetraPiprekova) [reply docs/functional/cart.md]: Changed

File: docs/functional/cart.md [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+- discount code can be created in the section pricing -> promo codes in the administration
+- the input for entering the discount code is in the cart on the Front-end
+- after entering the discount code in the cart, total price with VAT is recalculated using the button "Apply"
```
[the remark this answers is not in the document]
Comment (PetraPiprekova) [reply docs/functional/cart.md]: Thank you. I added this point.

File: docs/functional/cart.md (hunk 6 above)
Comment (PetraPiprekova) [reply docs/functional/cart.md]: The information about notification is added.

**Comment by PetraPiprekova** [comment]
Hi. I added new information, @LukasHeinz please check it.

**Comment by PetraPiprekova** [comment]
I updated the file. @LukasHeinz please, can you check it?

File: docs/functional/cart.md [hunk 12]
```diff
@@ -0,0 +1,37 @@
+# Cart
+This article describes functional behavior of a cart. Article shortly summarize its discount codes, behavior of cart content when user logs in and others.
+
+## Add to cart
+- a user can add the product to cart from the product detail or from the product listing (except for the product with variants)
+- after adding the product to the cart, a pop up window is displayed. There is information about adding the product to the cart together with the button “Go to cart”
```
Comment (LukasHeinz) [inline docs/functional/cart.md]:
```suggestion
- after adding the product to the cart, a pop up window is displayed. 
There is information about adding the product to the cart together with the button “Go to cart”.
When the product is no longer available there is information about that in pop up window.
```

File: docs/functional/cart.md [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+## Cart overview in header
+- shows the amount and total price with VAT of products in the cart
+- shows the information about empty cart
+- changes in content of cart in header are modified by Ajax
```
Comment (LukasHeinz) [inline docs/functional/cart.md]:
```suggestion
- shows current information about cart and changes even without page refresh - the update is done by AJAX
```

File: docs/functional/cart.md [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+## Page cart
+### URL
+- URL has format youreshopurl/cart (in English version)
```
Comment (LukasHeinz) [inline docs/functional/cart.md]:
```suggestion
- URL has format www.youreshopurl.com/cart (in English version)
```

File: docs/functional/cart.md [hunk 15]
```diff
@@ -0,0 +1,37 @@
+# Cart
+This article describes functional behavior of a cart. Article shortly summarize its discount codes, behavior of cart content when user logs in and others.
+
+## Add to cart
+- a user can add the product to cart from the product detail or from the product listing (except for the product with variants)
```
Comment (DavidOstrozlik) [inline docs/functional/cart.md]:
```suggestion
- a user can add the product to cart from the product detail or from the product listing (except for the product with variants as described in [another article](./behavior-of-product-variants.md))
```

File: docs/functional/cart.md [hunk 16]
```diff
@@ -0,0 +1,37 @@
+# Cart
+This article describes functional behavior of a cart. Article shortly summarize its discount codes, behavior of cart content when user logs in and others.
```
Comment (DavidOstrozlik) [inline docs/functional/cart.md]:
```suggestion
This article describes functional behavior of a cart. Article shortly summarizes its discount codes, behavior of cart content when user logs in and others.
```

File: docs/functional/cart.md [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+- after adding the product to the cart, a pop up window is displayed. There is information about adding the product to the cart together with the button “Go to cart”
+## Cart overview in header
+- shows the amount and total price with VAT of products in the cart
+- shows the information about empty cart
+- changes in content of cart in header are modified by Ajax
+## Page cart
+### URL
+- URL has format youreshopurl/cart (in English version)
+### The cart can be in these states:
+- **empty** - nothing was added to the cart yet, cart shows an image of empty cart and  the customer is informed that the cart is empty
```
Comment (DavidOstrozlik) [inline docs/functional/cart.md]:
```suggestion
- **empty** - nothing was added to the cart yet, cart shows an image of empty cart and the customer is informed that the cart is empty
```

File: docs/functional/cart.md [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+## Persistence
+- the cart, including its content, is stored in the database for:
+- 60 days for unregistered user
+- 120 days for registered user
+## Notification about product changes
+- if the product is added to the cart and its price is changed or is not available anymore, the user is informed of the changes
+- the information is displayed on any page after page refresh
+## Cart behavior when logging in, logging out and register the user
+- if an unregistered customer adds the product to the cart and makes the registration (the system logs the customer in) then the contents of the cart will be retained
+- if an unsigned customer adds the product to the cart and then logs in, the contents of the cart will be retained
```
Comment (DavidOstrozlik) [inline docs/functional/cart.md]:
```suggestion
- if an unsigned customer adds the product to the cart and then logs in, the contents of both unsigned and signed carts will merge
```

File: docs/functional/cart.md [hunk 19]
```diff
@@ -0,0 +1,37 @@
...
+- changes in content of cart in header are modified by Ajax
+## Page cart
+### URL
+- URL has format youreshopurl/cart (in English version)
+### The cart can be in these states:
+- **empty** - nothing was added to the cart yet, cart shows an image of empty cart and  the customer is informed that the cart is empty
+- **contains products** - the customer can see a list of products which have been added to the cart, customer can change quantity of products or remove product from the cart - these changes are modified by Ajax
+### Price
+- total price including VAT is automatically recalculated (without page refresh) if the product is removed or the amount is changed
+### Discount coupon
```
Comment (DavidOstrozlik) [inline docs/functional/cart.md]: Do we use `Discount coupon` or `Promo code`?

File: docs/functional/cart.md (hunk 15 above)
Comment (DavidOstrozlik) [inline docs/functional/cart.md]: Do we use `User` or `Customer`?

File: docs/functional/cart.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (DavidOstrozlik) [reply docs/functional/cart.md]: This is not true. If you add coupon the discount is added to the item so there is the same amount of items

File: docs/functional/cart.md [hunk 20]
```diff
@@ -0,0 +1,37 @@
...
+- the coupon is applied for every cart item individually so the number of items doubles
+- discount code can be removed from the cart and total price with VAT will be recalculated automatically
+## Order flow
+- the cart is the first step of order followed by a choice of shipping and payment
+## Persistence
+- the cart, including its content, is stored in the database for:
+- 60 days for unregistered user
+- 120 days for registered user
+## Notification about product changes
+- if the product is added to the cart and its price is changed or is not available anymore, the user is informed of the changes
```
Comment (DavidOstrozlik) [inline docs/functional/cart.md]:
```suggestion
- if the product is added to the cart and its price is changed or is not available anymore, the user is informed of the changes by a flash message
```

File: docs/functional/cart.md (hunk 19 above)
Comment (PetraPiprekova) [reply docs/functional/cart.md]: I changed discount code to promo code. Thanks

File: docs/functional/cart.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (PetraPiprekova) [reply docs/functional/cart.md]: Thank you David, it is true. Double amount of items is in order overview.

File: docs/functional/cart.md [hunk 21]
```diff
@@ -0,0 +1,39 @@
+# Cart
+This article describes functional behavior of a cart. Article shortly summarizes its promo codes, behavior of cart content when user logs in and others.
+
+## Add to cart
+- a user can add the product to cart from the product detail or from the product listing (except for the product with variants as described in [another article](./behavior-of-product-variants.md))
```
Comment (DavidOstrozlik) [inline docs/functional/cart.md]:
```suggestion
- a customer can add the product to cart from the product detail or from the product listing (except for the product with variants as described in [another article](./behavior-of-product-variants.md))
```

File: docs/functional/cart.md [hunk 22]
```diff
@@ -0,0 +1,39 @@
+# Cart
+This article describes functional behavior of a cart. Article shortly summarizes its promo codes, behavior of cart content when user logs in and others.
```
Comment (DavidOstrozlik) [inline docs/functional/cart.md]:
```suggestion
This article describes functional behavior of a cart. Article shortly summarizes its promo codes, behavior of cart content when customer logs in and others.
```

File: docs/functional/cart.md [hunk 23]
```diff
@@ -0,0 +1,39 @@
...
+### Promo code
+- the input for entering the promo code is in the cart on the Front-end
+- after entering the promo code in the cart, total price with VAT is recalculated using the button "Apply"
+- the coupon is applied for every cart item individually
+- promo code can be removed from the cart and total price with VAT will be recalculated automatically
+## Order flow
+- the cart is the first step of order followed by a choice of shipping and payment
+## Persistence
+- the cart, including its content, is stored in the database for:
+- 60 days for unregistered user
```
Comment (DavidOstrozlik) [inline docs/functional/cart.md]:
```suggestion
- 60 days for unregistered customer
```

File: docs/functional/cart.md [hunk 24]
```diff
@@ -0,0 +1,39 @@
...
+- 120 days for registered user
```
Comment (DavidOstrozlik) [inline docs/functional/cart.md]:
```suggestion
- 120 days for registered customer
```

File: docs/functional/cart.md [hunk 25]
```diff
@@ -0,0 +1,39 @@
...
+## Notification about product changes
+- if the product is added to the cart and its price is changed or is not available anymore, the user is informed of the changes by a flash message
```
Comment (DavidOstrozlik) [inline docs/functional/cart.md]:
```suggestion
- if the product is added to the cart and its price is changed or is not available anymore, the customer is informed of the changes by a flash message
```

File: docs/functional/cart.md [hunk 26]
```diff
@@ -0,0 +1,39 @@
...
+- the information is displayed on any page after page refresh
+## Cart behavior when logging in, logging out and register the user
```
Comment (DavidOstrozlik) [inline docs/functional/cart.md]:
```suggestion
## Cart behavior when logging in, logging out and register the customer
```

File: docs/functional/cart.md [hunk 27]
```diff
@@ -0,0 +1,39 @@
...
+- URL has format www.youreshopurl.com/cart (in English version)
+### The cart can be in these states:
+- **empty** - nothing was added to the cart yet, cart shows an image of empty cart and the customer is informed that the cart is empty
+- **contains products** - the customer can see a list of products which have been added to the cart, customer can change quantity of products or remove product from the cart - these changes are modified by Ajax
+### Price
+- total price including VAT is automatically recalculated (without page refresh) if the product is removed or the amount is changed
+### Promo code
+- the input for entering the promo code is in the cart on the Front-end
+- after entering the promo code in the cart, total price with VAT is recalculated using the button "Apply"
+- the coupon is applied for every cart item individually
```
Comment (DavidOstrozlik) [inline docs/functional/cart.md]:
```suggestion
- the discount is applied for every cart item individually
```

File: docs/functional/cart.md (hunk 22 above)
Comment (PetraPiprekova) [reply docs/functional/cart.md]: After agreement, I changed all customers to users.

**Pull request merged**